picard/util/checkupdate.py

```diff
@@ -0,0 +1,159 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from picard import (PICARD_VERSION, tagger, log, version_from_string, version_to_string, config)
+import picard.util.webbrowser2 as wb2
+from PyQt5 import QtCore
+from PyQt5.QtWidgets import QMessageBox
+from picard.util import load_json, compare_version_tuples
+from functools import partial
+import re
+import datetime
+
+
+# Used to strip leading and trailing text from version string.
+_RE_CLEAN_VERSION = re.compile('^[^0-9]*(.*)[^0-9]*$', re.IGNORECASE)
+
+
+# GitHub API information
+GITHUB_API = {
+    'host': 'api.github.com',
+    'port': 443,
+    'endpoint': {
+        'releases': '/repos/metabrainz/picard/releases',
+        'tags': '/repos/metabrainz/picard/git/refs/tags'
+    }
+}
+
+
+class UpdateCheckManager(QtCore.QObject):
+
+    def __init__(self):
+        super().__init__()
+
+        # Version tuple format:  str.key: ( str.version, ( int.major, int.minor, int.micro, str.type, int.development ), str.title, str.url )
+        self._available_versions = {
+            'stable': ('', (0, 0, 0, 'dev', 0), '', ''),
+            'beta': ('', (0, 0, 0, 'dev', 0), '', ''),
+            'dev': ('', (0, 0, 0, 'dev', 0), '', ''),
+        }
+        self._show_always = False
+        self._update_level = 'dev'
+
+    def check_update(self, show_always=False, update_level='dev', callback=None):
+        '''Checks if an update is available.
+
+        Compares the version number of the currently running instance of Picard
+        and displays a dialog box informing the user  if an update is available,
+        with an option of opening the Picard site in the browser to download the
+        update.  If there is no update available, no dialog will be shown unless
+        the "show_always" parameter has been set to True.  This allows for silent
+        checking during startup if so configured.
+
+        Args:
+            show_always: Boolean value indicating whether the results dialog
+                should be shown even when there is no update available.
+            update_level: Determines what type of updates to check.  If set to
+                'final' only stable release versions are checked.  If set to
+                'dev' both beta and stable releases are checked.
+            callback: Optional callback function.
+
+        Returns:
+            none.
+
+        Raises:
+            none.
+        '''
+        self._show_always = show_always
+        self._update_level = update_level
+
+        # Gets list of releases from GitHub website api.
+        output_text = _("Getting release information from GitHub.")
```

Comment:
This and below is exactly the code from `query_available_updates`, should call that instead of reimplementing it here,

Reply:
You're right.  That was a left over from a previous version and was supposed to have been removed.  I'll make the change.

Reply:
Change made, but I ended up having to force push it.  You should be able to see the corrected version in the current changes.